storage/src/journal/variable.rs

```diff
@@ -259,7 +267,7 @@ impl<E: Storage + Metrics, V: Codec> Journal<E, V> {
 
     /// Helper function to read an item from a [Read].
     async fn read_buffered(
-        reader: &mut Read<E::Blob>,
+        reader: &mut Read<Write<E::Blob>>,
```

Comment:
Not super familiar with the code, but seems like a strange pattern to have `Read<Write<E::Blob>>` instead of just `Read<E::Blob>` and `Write<E::Blob>`

Reply:
This is the cleanest way I've been able to organize this thus far (without the `journal` needing to know about any of the buffers being used under the hood and supporting distinct read/write buffers).

We could probably combine them as a `Buffered` blob (that supports both buffered write and replay) 🤔

Reply:
Created an issue for us to consider later: https://github.com/commonwarexyz/monorepo/issues/963

I "think" a `buffer::Blob` could make sense (could then do the same for `buffer::Sink` or something too). Not confident enough yet in the usage patterns but agree it looks a little odd.